libraries/Microsoft.Bot.Builder.Dialogs/Debugging/DebugSupport.cs

```diff
@@ -11,11 +11,53 @@ namespace Microsoft.Bot.Builder.Dialogs.Debugging
     /// </summary>
     public static class DebugSupport
     {
+        private static ISourceMap staticSourceMap = Debugging.SourceMap.Instance;
+        private static AsyncLocal<ISourceMap> asyncLocalSourceMap = new AsyncLocal<ISourceMap>();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to use async local values for 
+        /// <see cref="SourceMap"/>. This enables multiple concurrent operations 
+        /// to each use their own <see cref="ISourceMap"/>.
+        /// </summary>
+        /// <value>True if the <see cref="SourceMap"/> is local to the async 
+        /// context, otherwise false.</value>
+        public static bool UseAsyncLocal { get; set; }
+
         /// <summary>
         /// Gets or sets the source map instance.
         /// </summary>
         /// <value>The <see cref="SourceMap"/> instance.</value>
-        public static ISourceMap SourceMap { get; set; } = Debugging.SourceMap.Instance;
+        public static ISourceMap SourceMap
```

Comment:
Why change this class? Why not create AsyncSourceMap which internally has a AsyncLocal<T> which is emphermal?

Reply:
Ack -- we did that change -- I am okay not changing this code.